### PR TITLE
Move compile commands into the makefile

### PR DESCRIPTION
**Pull request by paulirish**
This change should…
- improve transparency: easier to see & compare the various build pipelines
- improve maintainability: easier to port & run the project to Windows

seem good?

### CODE REVIEW

**Comment by gauntface** [comment]
Looks good.

Only comment, why does each step have two cd's?

```
cd webpack; npm i;
cd webpack; time $(webpack) ../src/src/app.js ../src/dist/bundle.js
```

File: Makefile [hunk 1]
```diff
@@ -25,41 +35,51 @@ all:
 	make size
 
 typescript:
-	cd typescript; npm i; time npm run compile;
+	cd typescript; npm i;
+	cd typescript; time $(tsc) && $(browserify) -p bundle-collapser/plugin ./dist/*.js | $(uglifyjs) --compress --mangle --screw-ie8 - > ../src/dist/bundle.js
```
Comment (samccone) [inline Makefile:39]: does this one work? I am impressed that it is able to resolve `bundle-collapser/plugin` without node_modules

Comment (paulirish) [inline Makefile:39]: yup
image

**Comment by paulirish** [comment]
> Only comment, why does each step have two cd's?

because makefiles are weird like that. http://stackoverflow.com/a/1789616/89484

File: Makefile (hunk 1 above)
Comment (samccone) [inline Makefile:39]: nice theme :)

**Comment by samccone** [comment]
lgtm :+1: thx @paulirish

**Pull request merged**